crates/anonymization/src/core/hash.rs

```diff
@@ -0,0 +1,100 @@
+use argon2::Argon2;
+use base64::{engine::general_purpose, Engine as _};
+use sha2::{Digest, Sha256};
+use tiny_keccak::{Hasher as _, Sha3};
+
+use crate::{ano_error, core::AnoError};
+
+// Available hashing methods
+#[derive(PartialEq, Eq)]
+pub enum HashMethod {
+    /// Represents the SHA2 hash method with an optional salt.
+    SHA2(Option<Vec<u8>>),
+    /// Represents the SHA3 hash method with an optional salt.
+    SHA3(Option<Vec<u8>>),
+    /// Represents the Argon2 hash method with a mandatory salt.
+    Argon2(Vec<u8>),
+}
+
+impl HashMethod {
+    /// `HashMethod` constructor for interfaces
+    ///
+    /// * `method` - The hash method to use. This can be one of the following:
+    ///   SHA2, SHA3 or Argon2
+    /// * `salt` - An optional salt to use. Required with Argon2
+    pub fn new(hasher_method: &str, salt_opt: Option<Vec<u8>>) -> Result<Self, AnoError> {
+        match hasher_method {
+            "SHA2" => Ok(Self::SHA2(salt_opt)),
+            "SHA3" => Ok(Self::SHA3(salt_opt)),
+            "Argon2" => salt_opt.map_or_else(
+                || Err(ano_error!("Argon2 requires a salt value.")),
+                |salt| Ok(Self::Argon2(salt)),
+            ),
+            _ => Err(ano_error!("Not a valid hash method specified.")),
+        }
+    }
+}
+
+pub struct Hasher {
+    method: HashMethod, // The selected hash method
+}
```

Comment:
I changed this to avoid a check for the option salt  for `Argon2` in the apply function. The check is now done at the object init which is only called once, whereas the apply function will be called many times.